applications/TrilinosApplication/custom_processes/trilinos_levelset_convection_process.h

```diff
@@ -342,6 +316,65 @@ class TrilinosLevelSetConvectionProcess
     ///@name Private Operations
     ///@{
 
+    void InitializeConvectionStrategy(BuilderSolverPointerType pBuilderAndSolver)
+    {
+        KRATOS_TRY
+
+        // Get auxiliary member variables from base class
+        auto& r_base_model_part = BaseType::mrBaseModelPart;
+        const auto& r_level_set_var = *BaseType::mpLevelSetVar;
+        const auto& r_convect_var = *BaseType::mpConvectVar;
+
+        // Check the nodal database of the current partition
+        int n_nodes = r_base_model_part.NumberOfNodes();
+        int n_elems = r_base_model_part.NumberOfElements();
+
+        if (n_nodes > 0){
+            VariableUtils().CheckVariableExists<Variable<double>>(r_level_set_var, r_base_model_part.Nodes());
+            VariableUtils().CheckVariableExists<Variable<array_1d<double,3>>>(r_convect_var, r_base_model_part.Nodes());
+        }
+
+        // Check if the modelpart is globally empty
+        n_nodes = r_base_model_part.GetCommunicator().GetDataCommunicator().SumAll(n_nodes);
+        n_elems = r_base_model_part.GetCommunicator().GetDataCommunicator().SumAll(n_elems);
+        KRATOS_ERROR_IF(n_nodes == 0) << "The model has no nodes." << std::endl;
+        KRATOS_ERROR_IF(n_elems == 0) << "The model has no elements." << std::endl;
+
+        // Check if any partition has incorrect elements
+        if(TDim == 2){
+            int has_incorrect_elems = r_base_model_part.NumberOfElements() ? r_base_model_part.ElementsBegin()->GetGeometry().GetGeometryFamily() != GeometryData::Kratos_Triangle : 0;
+            has_incorrect_elems = r_base_model_part.GetCommunicator().GetDataCommunicator().SumAll(has_incorrect_elems);
```

Comment:
I don't think it is necessary to throw on every rank
MPI will anyway terminate if errors on some ranks only

hence you can skip this check which will improve performance as less communication

Reply:
Makes sense. I'll use a bool at each rank.

Reply:
Done.